### ISSUE TITLE
TP11 ej 15 movetofrontADT

### ISSUE THREAD

**Issue by valensangui8** [marcelogarberoglio/PI#288]
Buenas Marcelo, como va? 
tengo una duda de si esta bien hacer lo que hice o no en la función get del adt de movetofront.

static Tlist search(moveToFrontADT moveToFront, Tlist list, elemType elem, int (*cmp) (elemType e1, elemType e2),  size_t * flag){
    if(list==NULL){
        *flag = 0;
        return list;
    }
    if(cmp(list->elem, elem)==0){
        Tlist aux = list;
        list = list->tail;
        aux->tail = moveToFront->first;
        *flag = 1;
        return aux;
    }
    list->tail = search(moveToFront, list->tail, elem, cmp, flag);
    return list;
} 

/* Retorna una copia del elemento. Si no existe retorna NULL.
** Para saber si el elemento está, usa la función compare. 
** Si el elemento estaba lo ubica al principio.
*/
elemType * get(moveToFrontADT moveToFront, elemType elem){
    size_t flag;
    moveToFront->first = search(moveToFront,moveToFront->first,elem,moveToFront->compare,&flag);
    if(flag == 0){
        return NULL;
    }
    elemType * ans = malloc(sizeof(*ans));
    *ans = moveToFront->first->elem;
    return ans;
}

**Comment by valensangui8** [comment]
te dejo tambien la estructura para que la tengas.

struct node{
    elemType elem;
    struct node * tail;
};

typedef struct node * Tlist;

struct moveToFrontCDT{
    Tlist first;
    Tlist iter;
    size_t size;
    int (*compare) (elemType e1, elemType e2);
};

**Comment by marcelogarberoglio** [comment]
No, no está bien, ya que search aunque lo encuentre igual devuelve el primero de la lista. Cuando search lo encuentre y retorna aux, ese aux se asigna en 
    list->tail = search(moveToFront, list->tail, elem, cmp, flag);
y como valía lisst, que es lo que recibió, sigue siendo el mismo list->tail.
Pero además hiciste que el siguiente apunte a first, por lo que en realidad las consecuencias de esta función es que perdés referencia a todos los nodos que seguían al buscado, y te queda una lista circular
Fijate de hacer el esquema de una lista y ver cómo van quedando los punteros


Adem;as. hacer
    list = list->tail;
sólo cambia una varibale local. Esa proposición se puede sacar y es lo mismo.
